apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -149,11 +149,13 @@ internal RetryInfo(DateTime firstTimeOfFailureInUtc, TimeSpan? backoff = null)
         private readonly CancellationTokenSource _pushStopCts;
         private readonly CancellationTokenSource _softStopCts;
         private readonly SubscriptionName _subscriptionName;
-        private readonly int _modifyDeadlineSeconds; // Seconds to add to deadling on lease extension.
-        private readonly TimeSpan _autoExtendInterval; // Delay between auto-extends for non-exactly once delivery subscriptions.
-        private readonly TimeSpan _autoExtendIntervalForExactlyOnceDelivery; // Delay between auto-extends for exactly once delivery subscriptions.
-        private readonly TimeSpan _maxExtensionDuration; // Maximum duration for which a message lease will be extended.
+        private readonly int _modifyDeadlineSeconds; // Seconds to add to deadline on lease extension.
```

Comment:
Given that we've got three pairs of fields, perhaps we should have a type of LeaseExtensionSettings or similar? (That could be done later.)

Reply:
Sure Amanda.  I am very happy to change.

Here are my thoughts:

- For non-eod subs, ackDeadline is user provided AckDeadline or the default (which happens to be 60 seconds)

- For eod subs, ackDeadline is user provided AckDeadline or the minimum Ack Deadline (which happens to be 60 seconds).

- Earlier, because of the above defaults, I had reused the _modifyAckDeadline across both the subs. With both the defaults matching, we do not require the restart of subscriber client even if sub type is changed mid-flight.

- While prototyping this fix, I felt if tomorrow (this is hypothetical and violates YAGNI) the MinimumAckDeadline is changed to say 75 seconds and we have default AckDeadline as 60 seconds, then we will have different values of AckDeadline for eod and non-eod subs and so reusing _modifyAckDeadline will require subscriber to restart the client. Keeping them separate prevents the need to restart. 

- We have had an issue reported in which subscription is changed in-flight while the client is running. This was when the eod subs was first introduced. 

Please let me know what you think.